pkg/clients/database/rds.go

```diff
@@ -49,6 +49,11 @@ const (
 	errGetPasswordSecretFailed = "cannot get password secret"
 )
 
+type Cache struct {
+	AddTags    []rdstypes.Tag
+	RemoveTags []string
+}
+
```

Comment:
The `Cache` struct is controller specific and should be in `/pkg/controller`.

Reply:
I put the `Cache` struct into `pkg/clients/rds.go` so it could be used in `IsUpToDate()`. Due to the controller and client being in different packages and the controller needing to import the client ("import cycle not allowed"), this was the only way I found for this.

But I will try to find another solution.

Reply:
Ok, simply moved the `Cache` struct now into the controller.

Maybe the "cache" solution works/writes best in ACK controller setups/resources.
A non-cache solution, I found in other "hand-written" controller setups, is to call `DiffTag` again when in `Update()`:
https://github.com/crossplane-contrib/provider-aws/blob/4124fb7ccb72938dcfe5ada372d910f81884a270/pkg/controller/acm/controller.go#L194